### PR TITLE
Integrate llvm/llvm-project@c54064de80e93494d1d44550b56ce8f2f3cf9c4b

### PR DESCRIPTION
**Pull request by Max191**
Bump LLVM to include https://github.com/llvm/llvm-project/commit/205dce6029bed302f354c0bde5d8c5804f214051 and https://github.com/llvm/llvm-project/commit/3f18f6a2cfecb080f006477c46d3626102841a17

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp [hunk 1]
```diff
@@ -398,7 +398,15 @@ struct FusionOfTensorOpsPass
...
+              return producerGenericOp->hasOneUse() &&
+                     llvm::all_of(producerGenericOp.getIteratorTypesArray(),
+                                  linalg::isParallelIterator);
+            }
+
+            if (auto consumerGenericOp =
+                    dyn_cast<linalg::GenericOp>(producer)) {
+              return llvm::all_of(consumerGenericOp.getIteratorTypesArray(),
+                                  linalg::isParallelIterator);
             }
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: Is it broken by https://github.com/llvm/llvm-project/commit/3f18f6a2cfecb080f006477c46d3626102841a17 ?

If so, @qedawkins could you help review?

Comment (qedawkins) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: Yeah, this is intended actually. The changes upstream just expand what the pattern is capable of, but I don't think we want to do the kinds of fusions there by default just yet.

@Max191 this also needs to check that either the producer or consumer is a generic (we don't want to enable fusion-by-expansion of named ops just yet I don't believe).

Comment (Max191) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: I talked to Quinn about this, and we figured we should start without changing anything, and we can try turning it on later and see if anything breaks.

Comment (qedawkins) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: yeah what Max said. The plan is to actually use these patterns in another place first which will use a different control function.

Comment (hanhanW) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: SGTM, can we update the comment as well?

Comment (hanhanW) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp]: cc @MaheshRavishankar who is the codeowner

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp [hunk 2]
```diff
@@ -395,10 +395,25 @@ struct FusionOfTensorOpsPass
...
+            // Do not fuse with other producer or consumer linalg ops if they
+            // have any reduction iterators.
+            if (auto producerLinalgOp = dyn_cast<linalg::LinalgOp>(producer)) {
+              return llvm::all_of(producerLinalgOp.getIteratorTypesArray(),
+                                  linalg::isParallelIterator);
+            }
+
+            if (auto consumerLinalgOp = dyn_cast<linalg::LinalgOp>(consumer)) {
+              return llvm::all_of(consumerLinalgOp.getIteratorTypesArray(),
+                                  linalg::isParallelIterator);
```
Comment (qedawkins) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp:416]: Add a note that this also prevents any of the named ops we care about from being expanded.

**Review by qedawkins — APPROVED** [review]

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp (hunk 2 above)
Comment (hanhanW) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp:416]: +1

**Review by hanhanW — APPROVED** [review]

**Pull request merged**